File: src/main/java/com/addressbook/SessionListener.java
package com.addressbook;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import javax.servlet.http.HttpSessionEvent;
import javax.servlet.http.HttpSessionListener;
import java.util.concurrent.atomic.AtomicInteger;

public class SessionListener implements HttpSessionListener {

    private static Logger logger = LoggerFactory.getLogger(SessionListener.class);

    private final AtomicInteger activeSessions;

    public SessionListener() {
        activeSessions = new AtomicInteger();
    }

    public void sessionCreated(final HttpSessionEvent event) {
        activeSessions.incrementAndGet();
        logger.info("HttpSession created: " + event.getSession().getId());
        logger.info("Total sessions: " + activeSessions.get());
    }

    public void sessionDestroyed(final HttpSessionEvent event) {
        activeSessions.decrementAndGet();
        logger.info("HttpSession destroyed: " + event.getSession().getId());
        logger.info("Total sessions: " + activeSessions.get());
    }
}

File: addressbook-main/src/main/kotlin/com/addressbook/configuration/SessionListener.kt
package com.addressbook.configuration

import org.slf4j.LoggerFactory
import java.util.concurrent.atomic.AtomicInteger
import javax.servlet.http.HttpSessionEvent
import javax.servlet.http.HttpSessionListener

class SessionListener : HttpSessionListener {

    private val logger = LoggerFactory.getLogger(SessionListener::class.java)
    private val activeSessions: AtomicInteger = AtomicInteger()

    override fun sessionCreated(event: HttpSessionEvent) {
        activeSessions.incrementAndGet()
        logger.info("HttpSession created: {}", event.session.id)
        logger.info("Total sessions: {}", activeSessions.get())
    }

    override fun sessionDestroyed(event: HttpSessionEvent) {
        activeSessions.decrementAndGet()
        logger.info("HttpSession destroyed: {}", event.session.id)
        logger.info("Total sessions: {}", activeSessions.get())
    }
}

